packages/rocketchat-theme/server/server.js

```diff
@@ -0,0 +1,189 @@
+	if (path === (`${ prefix }/__cordova/theme.css`) || path === (`${ prefix }/theme.css`)) {
+		css = RocketChat.theme.getCss();
+		hash = crypto.createHash('sha1').update(css).digest('hex');
+		res.setHeader('Content-Type', 'text/css; charset=UTF-8');
+		res.setHeader('ETag', `"${ hash }"`);
+		res.write(css);
+		return res.end();
+	} else {
+		return next();
+	}
+});
+
+const calculateClientHash = WebAppHashing.calculateClientHash;
+
+WebAppHashing.calculateClientHash = function(manifest, includeFilter, runtimeConfigOverride) {
+	let hash;
+	let themeManifestItem;
+	const css = RocketChat.theme.getCss();
+	if (css.trim() !== '') {
+		hash = crypto.createHash('sha1').update(css).digest('hex');
+		themeManifestItem = _.find(manifest, function(item) {
+			return item.path === 'app/theme.css';
+		});
+		if (themeManifestItem == null) {
+			themeManifestItem = {};
+			manifest.push(themeManifestItem);
+		}
+		themeManifestItem.path = 'app/theme.css';
+		themeManifestItem.type = 'css';
+		themeManifestItem.cacheable = true;
+		themeManifestItem.where = 'client';
+		themeManifestItem.url = `/theme.css?${ hash }`;
+		themeManifestItem.size = css.length;
+		themeManifestItem.hash = hash;
+	}
+	return calculateClientHash.call(this, manifest, includeFilter, runtimeConfigOverride);
+};
+
+RocketChat.theme = new class {
+	constructor() {
+		this.variables = {};
+		this.packageCallbacks = [];
+		this.files = ['server/lesshat.less', 'server/colors.less'];
+		this.customCSS = '';
+		RocketChat.settings.add('css', '');
+		RocketChat.settings.addGroup('Layout');
+		RocketChat.settings.onload('css', Meteor.bindEnvironment(() => {
+			return function(key, value, initialLoad) {
+				if (!initialLoad) {
+					return Meteor.startup(function() {
+						return process.emit('message', {
+							refresh: 'client'
+						});
+					});
+				}
+			};
+		})(this));
+		this.compileDelayed = _.debounce(Meteor.bindEnvironment(this.compile.bind(this)), 100);
+		Meteor.startup(() => {
+			return function() {
+				return RocketChat.settings.onAfterInitialLoad(function() {
+					return RocketChat.settings.get('*', Meteor.bindEnvironment(function(key, value) {
+						let name;
+						if (key === 'theme-custom-css' && value != null) {
+							this.customCSS = value;
+						} else if (/^theme-.+/.test(key) === true) {
+							name = key.replace(/^theme-[a-z]+-/, '');
+							if (this.variables[name] != null) {
+								this.variables[name].value = value;
+							}
+						} else {
+							return;
+						}
+						return this.compileDelayed();
+					}));
+				});
+			};
+		});
+	}
+
+	compile() {
+
+		let content = [this.getVariablesAsLess(), ...Object.keys(this.files || []).map((key) => {
+			const file = this.files[key];
+			return Assets.getText(file);
+		}), ...Object.keys(this.packageCallbacks || []).map((key)=>{
+			const value = this.packageCallback[key];
+			return value();
+		}).filter(value => _.isString(value))];
```

Comment:
Please, split this code in parts.

this.files is not an object, it's and array, this.packageCallback is an array as well